src/gateways/S3Gateway.ts

```diff
@@ -74,6 +74,37 @@ export class S3Gateway {
     return JSON.parse(object.Body.toString());
   }
 
+  async deleteByDocumentId(documentId: string): Promise<void> {
+    this.logger
+      .mergeContext({ documentId })
+      .log('[s3] removing all objects associated with id');
+
+    const listing = await this.client.listObjects({
+      Bucket: this.bucketName,
+      Prefix: `${documentId}/`,
+      MaxKeys: 2, // safeguard against bad prefixes
```

Comment:
This way we only ever delete at most 2 things, so we can't accidentally delete the entire contents of the bucket if there is an error with the prefixing or something.

Reply:
So, there can only ever be 2 files in a folder - metadata and the doc right?

Reply:
Yeah, at least at the moment - if we ever start storing more stuff will need to make sure this is also updated.